src/hugr/views/sibling.rs

```diff
@@ -0,0 +1,572 @@
+            .filter_map(|(n, p)| self.base.get_optype(n).signature().get(p))
+            .map(|edgekind| {
+                if let EdgeKind::Value(t) = edgekind {
+                    t
+                } else {
+                    unimplemented!(
+                        "ControlFlow, StateOrder and Static edges not supported for output"
+                    )
+                }
+            })
+            .collect_vec();
+        AbstractSignature::new(input, output, input_static)
+    }
+
+    /// The parent of the sibling subgraph.
+    pub fn get_parent(&self) -> Node
+    where
+        Base: HugrView,
+    {
+        self.base
+            .get_parent(self.nodes[0])
+            .expect("invalid subgraph")
+    }
+
+    /// Construct a [`SimpleReplacement`] to replace `self` with `replacement`.
+    ///
+    /// `replacement` must be a hugr with DFG root and its signature must
+    /// match the signature of the subgraph.
+    ///
+    /// We currently do not support inputs of the replacement graph being
+    /// copied.
+    ///
+    /// May return one of the following five errors
+    ///  - [`InvalidReplacement::InvalidDataflowGraph`]: the replacement
+    ///    graph is not a [`crate::ops::OpTag::DataflowParent`]-rooted graph,
+    ///  - [`InvalidReplacement::InvalidDataflowParent`]: the replacement does
+    ///    not have an input and output node,
+    ///  - [`InvalidReplacement::InvalidSignature`]: the signature of the
+    ///    replacement DFG does not match the subgraph signature,
+    ///  - [`InvalidReplacement::NonConvexSubgrah`]: the sibling subgraph is not
+    ///    convex, or
+    ///  - [`InvalidReplacement::InvalidCopy`]: the replacement has a copy at
+    ///    the input boundary.
+    pub fn create_simple_replacement(
+        &self,
+        replacement: Hugr,
+    ) -> Result<SimpleReplacement, InvalidReplacement>
+    where
+        Base: HugrView,
+    {
+        let removal = self.nodes().iter().copied().collect();
+
+        let rep_root = replacement.root();
+        let dfg_optype = replacement.get_optype(rep_root);
+        if dfg_optype.tag() != OpTag::Dfg {
+            return Err(InvalidReplacement::InvalidDataflowGraph);
+        }
+        let Some((rep_input, rep_output)) = replacement
+            .children(rep_root)
+            .take(2)
+            .collect_tuple()
+        else { return Err(InvalidReplacement::InvalidDataflowParent) };
+        if dfg_optype.signature() != self.signature() {
+            return Err(InvalidReplacement::InvalidSignature);
+        }
+        let rep_inputs = replacement
+            .node_outputs(rep_input)
+            .filter(|&p| {
+                replacement
+                    .get_optype(rep_input)
+                    .signature()
+                    .get(p)
+                    .is_some()
+            })
+            .map(|p| {
+                replacement
+                    .linked_ports(rep_input, p)
+                    .exactly_one()
```

Comment:
If the replacement copies one of its inputs, won't this fire?

Reply:
Ah I see that's exactly what `InvalidCopy` is about. Hmmm. @cqc-alec do we need to extend/tweak SimpleReplace a bit here, to allow the replacement to copy its inputs?

Reply:
Sorry @acl-cqc , I'm confused here. Yes, it breaks if the replacement copies one of its inputs, but that is exactly what line 361 handles, and returns a `InvalidReplacement::InvalidCopy` error. At the output boundary, this problem does not exist because two edges cannot connect to the same port on the output port. If there was a mismatch in the output boundary it would get caught by the signature check on line 345.

Am I missing your point?

Reply:
Yes, indeed, you are quite right. (Crossed in post?)

Reply:
Right, chatted to Alec, think SimpleReplacement is fine (i.e. capable) as it stands. What I meant to say was, you can handle this case, and it's no more difficult than what you do for the outputs.
I think you want:
```
let rep_inputs = replacement // as before
    ... // filter as before
    .map(|p| replacement.linked_ports(rep_input, p).collect()) // in place of the map+exactly_one+error you had before
   .collect::<Vec<Vec<_>>>();
...
let nu_inp = rep_inputs
   .into_iter()
   .zip_eq(incoming)
   .flat_map(|(rep_inp, inc)| {
      rep_inp
         .into_iter()
         .map(|tgt| (tgt, inc))
         .collect::<Vec<_>>()
      })
   .collect();
```

Reply:
Oh very nice!!